### ISSUE TITLE
Should unsupported-base warning be triggered when inheriting from a type?

### ISSUE THREAD

**Issue by athoscouto** [astral-sh/ty#3952]
### Question

With the following code

```python
class A:
    pass


def get_a_flavor() -> type[A]:
    class B(A):
        pass

    return B


B = get_a_flavor()


class C(B):
    pass
```

I get:

```lua
warning[unsupported-base]: Unsupported class base
  --> src/test.py:15:9
   |
15 | class C(B):
   |         ^ Has type `type[A]`
16 |     pass
   |
info: ty cannot resolve a consistent method resolution order (MRO) for class `C` due to this base
info: Only class objects or `Any` are supported as class bases


Documentation: https://ty.dev/rules#unsupported-base
```

Should `ty` support this setup?

--- 

`SQLModel` subclasses take optional `metaclass` and `registry` parameters and I'm trying to use a function to reduce boilerplate when picking those paramters.

### Version

ty 0.0.56 (3078f9fa2 2026-07-01)

**Comment by carljm** [comment]
Hi, thanks for the report! This is the same as #1324. It looks like pyrefly and pyright accept this inheritance, but that's not really correct/sound: they will treat `C` as if it inherits directly from `A`, which it does not, and they'll entirely ignore the existence of the actual class `B`, and this can lead a type checker to incorrect conclusions and false negatives. For example, `B` could narrow a type from `A`, and then `C` would not see that, leading to runtime errors the type checker cannot catch.
